core/drivers/wdt/watchdog_sm.c

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2022 Microchip
+ */
+
+#include <drivers/wdt.h>
+#include <sm/optee_smc.h>
+#include <sm/psci.h>
+
+enum smcwd_call {
+	SMCWD_INIT		= 0,
+	SMCWD_SET_TIMEOUT	= 1,
+	SMCWD_ENABLE		= 2,
+	SMCWD_PET		= 3,
+	SMCWD_GET_TIMELEFT	= 4,
+};
+
+static unsigned long wdt_min_timeout;
+static unsigned long wdt_max_timeout;
+
+enum sm_handler_ret __wdt_sm_handler(struct thread_smc_args *args)
+{
+	switch (args->a1) {
```

Comment:
What happens if this function is called concurrently on two different CPUs?

Reply:
Main concern about that is `wdt_min_timeout` and `wdt_max_timeout` values that are potentially accessed concurrently even if the same values will be returned. I'm not sure where the concurrency should be handled. From what I already saw in OP-TEE, it seems you prefer to handle this in drivers but I'm not entirely sure.

Reply:
`wdt_min_timeout` and `wdt_max_timeout` must be protected here since they are read in this function. Other aspects should probably be handled inside the driver in case that driver is accessed somewhere else too.